components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -294,9 +350,19 @@ void Nudging::run_impl (const double dt)
       Kokkos::deep_copy(atm_state_view,int_state_view);
     } else {
       // Back out a tendency and apply it.
-      apply_tendency(atm_state_field, int_state_field, dt);
+      if (m_use_weights <= 0) {
+        apply_tendency(atm_state_field, int_state_field, dt);
+      } else {
+        // get nudging weights field
```

Comment:
Question, do we apply weights by column or is it fully 3D?  If it's by column then we shouldn't need to vertically interpolate the weights as you suggested.

Reply:
Ahh okay, that makes sense.  Then I would think we need to vertically interpolate the weights if we are already vertically interpolating the actual nudging data, to be consistent.  Also, one could imagine nudging data that is on a different vertical grid than the simulation, for example 72 vs. 128 levels.

Reply:
sure, we can call the online mapping function to do the interpolation on vertical grid

Reply:
My vision was that the user would have to supply nudging weights on exactly the same grid as used for the physics parameterizations in the target simulation - no interpolation in horizontal, vertical or time (because the same weights would be used for all times). My logic is that it should be super-trivial to create a weights file for your target grid (just open an output file with a 3d output on that grid and the lat/lon dimensions, create a variable of the same size minus the time dimension, then loop over all the columns and assign a weight based on its lat/lon... Easier to create a new weights file than to add code for interpolation.

Reply:
@PeterCaldwell, sure, the code is working in the way you described at the moment

Reply:
I think this issue can be resolved by just deleting the comment lines directly below.